time_stamper_cv/src/ConvexShape.cpp

```diff
@@ -0,0 +1,135 @@
+#include "ConvexShape.h"
+#include <utility>
+#include "opencv2/opencv.hpp"
+#include "Trigonometry.h"
+
+ConvexShape::ConvexShape(int tolerance)
+: tolerance_(tolerance) {}
+
+void ConvexShape::Process(PointVector raw_points) {
+  hull_.clear();
+  point_angles_sorted_.clear();
+  raw_points_ = std::move(raw_points);
+
+  calculateHull();
+  if (isHullValid()) {
+    calculateSortedPointAngles();
+  }
+}
+
+bool ConvexShape::ToleranceFilter(int expected_value, double actual_value) const {
```

Comment:
int vs double

Reply:
naming: Filter in robotics is very often used, so better something about "InRange" or "Bounded"